src/integrations.py

```diff
@@ -98,3 +120,132 @@ def auxiliary_data(self) -> AuxiliaryData:
+
+        hostname = charm.config.get("hostname")
+        self.cert_handler = CertHandler(
+            charm,
+            key="glauth-server-cert",
+            peer_relation_name="glauth-peers",
+            cert_subject=hostname,
+            extra_sans_dns=[hostname],
+        )
+
+    @property
+    def _ca_cert(self) -> Optional[str]:
+        return self.cert_handler.ca
+
+    @property
+    def _server_key(self) -> Optional[str]:
+        return self.cert_handler.key
+
+    @property
+    def _server_cert(self) -> Optional[str]:
+        return self.cert_handler.cert
+
+    @property
+    def _ca_chain(self) -> list[str]:
+        return self.cert_handler.chain
+
+    @property
+    def cert_data(self) -> CertificateData:
+        return CertificateData(
+            ca_cert=self._ca_cert,
+            ca_chain=self._ca_chain,
+            cert=self._server_cert,
+        )
+
+    def update_certificates(self) -> None:
+        if not self.cert_handler.enabled:
+            logger.debug("The certificates integration is not ready.")
+            self._remove_certificates()
+            return
+
+        if not self.certs_ready():
+            logger.debug("The certificates data is not ready.")
+            self._remove_certificates()
+            return
+
+        self._prepare_certificates()
+        self._add_certificates()
+
+    def certs_ready(self) -> bool:
+        return all((self._ca_cert, self._ca_chain, self._server_key, self._server_cert))
+
+    def _prepare_certificates(self) -> None:
+        SERVER_CA_CERT.write_text(self._ca_cert)  # type: ignore[arg-type]
+        SERVER_KEY.write_text(self._server_key)  # type: ignore[arg-type]
+        SERVER_CERT.write_text(self._server_cert)  # type: ignore[arg-type]
+
+        try:
+            for attempt in Retrying(
+                wait=wait_fixed(3),
+                stop=stop_after_attempt(3),
+                retry=retry_if_exception_type(subprocess.CalledProcessError),
+                reraise=True,
+            ):
+                with attempt:
+                    subprocess.run(
+                        ["update-ca-certificates", "--fresh"],
+                        check=True,
+                        text=True,
+                        capture_output=True,
+                    )
+        except subprocess.CalledProcessError as e:
+            logger.error(f"{e.stderr}")
+
+    def _add_certificates(self) -> None:
+        self._container.push(CERTIFICATE_FILE, CERTIFICATE_FILE.read_text(), make_dirs=True)
+        self._container.push(SERVER_CA_CERT, self._ca_cert, make_dirs=True)
+        self._container.push(SERVER_KEY, self._server_key, make_dirs=True)
+        self._container.push(SERVER_CERT, self._server_cert, make_dirs=True)
+
+    def _remove_certificates(self) -> None:
+        with suppress(PathError):
+            for file in (CERTIFICATE_FILE, SERVER_CA_CERT, SERVER_KEY, SERVER_CERT):
+                self._container.remove_path(file)
```

Comment:
If one removing one file fails, should the function exit or should it try for the rest? 

I think that it is safer to put the suppress in the `for` loop, so that we make sure all files will be removed.